dockerfiles/init/manifests/codenvy.env

```diff
@@ -86,17 +86,121 @@
 #####                                                                              #####
-#####                                   DOCKER                                     #####
+#####                                 NETWORKING                                   #####
 #####                                                                              #####
-# IP Address
-#     The IP address of the Docker daemon that is running on your host. We do a 
-#     self-discovery to set this. You can combine this with values of DOCKER_HOST to 
-#     change communications from socket to TCP.
+#
+# Affects how the Codenvy server connects to the Docker Daemon, how the Che server and
+# workspaces establish connections to one another, and also how remote browser clients
+# should discover and connect to workspaces.
+#
+# Codenvy goes through a progression algorithm to establish the protocol, IP address and
+# port to establish communications with it is booting or starting a workspace.
+#
+# Browser --> Codenvy Server
+#    1. Default is 'http://localhost:${SERVER_PORT}/wsmaster/api'.
+#    2. Else use the value of CODENVY_API
+#
+# Codenvy Server --> Docker Daemon Progression:
+#    1. Use the value of CODENVY_DOCKER_DAEMON__URL
+#    2. Else, use the value of DOCKER_HOST system variable
+#    3. Else, use Unix socket over unix:///var/run/docker.sock
+#
+# Codenvy Server --> Workspace Connection:
+#    - If CODENVY_DOCKER_SERVER__EVALUATION__STRATEGY is 'default':
+#        1. Use the value of CODENVY_DOCKER_IP
+#        2. Else, if server connects over Unix socket, then use localhost
+#        3. Else, use DOCKER_HOST
+#    - If CODENVY_DOCKER_SERVER__EVALUATION__STRATEGY is 'docker-local':
+#        1. Use the address of the workspace container within the docker network 
+#           and exposed ports
+#        2. If address is missing, if server connects over Unix socket, then use 
+#           localhost and exposed ports
+#        3. Else, use DOCKER_HOST and published ports
+#
+# Browser --> Workspace Connection:
+#    - If CODENVY_DOCKER_SERVER__EVALUATION__STRATEGY is 'default':
+#        1. If set use the value of CODENVY_DOCKER_IP_EXTERNAL
+#        2. Else if set use the value of CODENVY_DOCKER_IP
+#        3. Else, if server connects over Unix socket, then use localhost
+#        4. Else, use DOCKER_HOST
+#    - If CODENVY_DOCKER_SERVER__EVALUATION__STRATEGY is 'docker-local':
+#        1. If set use the value of CODENVY_DOCKER_IP_EXTERNAL
+#        2. Else use the address of the workspace container within the docker network, 
+#           if it is set
+#        3. If address is missing, if server connects over Unix socket, then use 
+#           localhost
+#        4. Else, use DOCKER_HOST
+#
+# Workspace Agent --> Codenvy Server
+#    1. Default is 'http://codenvy-host:${SERVER_PORT}/wsmaster/api', where 
+#       'codenvy-host' is IP of server.
+#    2. Else, use value of CODENVY_WORKSPACE_CHE__SERVER__ENDPOINT
```

Comment:
Starting from default is confusing to me. I would prefer to read docs that says:
1. If <> is set then
1.1. If <> is ....
1.2 Otherwise ...
2. Otherwise default value <> is used.